tests/test_benchmark_runner.py

```diff
@@ -0,0 +1,320 @@
+
+
+class TestRunners:
+    @pytest.fixture(scope="class", autouse=True)
+    def load_plugins(self):
+        load_plugins()
+
+    @pytest.fixture()
+    def item_from_test(self):
+        return TestItem(prompts=[PromptWithContext(prompt=TextPrompt(text="Hello!"), source_id="hello")])
+
+    @pytest.fixture()
+    def a_test(self, item_from_test, fake_secrets):
+        return ATest("a_test", [item_from_test], fake_secrets)
+
+    @pytest.fixture()
+    def a_sut(self):
+        return ModelGaugeSut("demo_yes_no")
+
+    @pytest.fixture()
+    def exploding_sut(self, a_sut):
+        real_sut = MagicMock()
+        real_sut.evaluate.side_effect = ValueError("sut done broke")
+        a_sut.instance = lambda _: real_sut
+        return a_sut
+
+    @pytest.fixture()
+    def a_wrapped_test(self, a_test, tmp_path):
+        return ModelgaugeTestWrapper(a_test, tmp_path)
+
+    @pytest.fixture()
+    def exploding_wrapped_test(self, item_from_test, tmp_path):
+        a = MagicMock(spec=DemoYBadAnnotator)
+        a.annotate.side_effect = ValueError("annotator done broke")
+        raw_test = ATest("a_test", [item_from_test], None, annotator=a)
+        return ModelgaugeTestWrapper(raw_test, tmp_path)
+
+    @pytest.fixture()
+    def benchmark(self, a_test):
+        class AHazard(HazardDefinition):
+            def tests(self, secrets: RawSecrets) -> List[PromptResponseTest]:
+                return [a_test]
+
+            def score(self, sut_scores: Mapping[str, TestRecord]) -> HazardScore:
+                count = 0
+                total = 0
+                for key, value in sut_scores.items():
+                    d = value.result.data
+                    count += d["badness_count"]
+                    total += d["total_badness"]
+
+                score = ValueEstimate.make(total / count, count)
+
+                test_scores = {}
+                return HazardScore(hazard_definition=self, score=score, test_scores=test_scores)
+
+        class ABenchmark(BenchmarkDefinition):
+            def _make_hazards(self) -> Sequence[HazardDefinition]:
+                return [AHazard()]
+
+        return ABenchmark()
+
+    def a_run(self, tmp_path, **kwargs) -> BenchmarkRun:
+        runner = BenchmarkRunner(tmp_path / "run")
+        for key, value in kwargs.items():
+            runner.__dict__[key] = value
+        if runner.secrets is None:
+            runner.secrets = fake_all_secrets()
+        return BenchmarkRun(runner)
+
+    @pytest.fixture()
+    def sut_response(self):
+        return SUTResponse(completions=[SUTCompletion(text="Hello, is it me you're looking for?")])
+
+    @pytest.fixture()
+    def hazard(self):
+        pass
+
+    def test_benchmark_source(self, fake_secrets, tmp_path, benchmark):
+        bsa = TestRunItemSource(self.a_run(tmp_path, secrets=fake_secrets, max_items=1, benchmarks=[benchmark]))
+        iterator = iter(bsa.new_item_iterable())
+        first_item = next(iterator)
+        print(first_item)
```

Comment:
`pytest` will eat anything that gets `print`ed. Should we use logging here instead?

Reply:
Leftover debugging code. Removed!